azurerm/helpers/azure/batch_pool.go

```diff
@@ -553,3 +553,162 @@ func FlattenBatchMetaData(metadatas *[]batch.MetadataItem) map[string]interface{
+func ExpandBatchPoolEndpointConfiguration(list []interface{}) (*batch.PoolEndpointConfiguration, error) {
+	if len(list) == 0 {
+		return nil, nil
+	}
+
+	endpointConfigurationValue := list[0].(map[string]interface{})
+
+	endpointConfiguration := &batch.PoolEndpointConfiguration{}
+
+	inboundNatPoolsValues := endpointConfigurationValue["inbound_nat_pools"].([]interface{})
+	inboundNatPools := make([]batch.InboundNatPool, len(inboundNatPoolsValues))
+
+	for i, inboundNatPoolsValue := range inboundNatPoolsValues {
+		inboundNatPool := inboundNatPoolsValue.(map[string]interface{})
+
+		name := inboundNatPool["name"].(string)
+		protocol := batch.InboundEndpointProtocol(inboundNatPool["protocol"].(string))
+		backendPort := int32(inboundNatPool["backend_port"].(int))
+		frontendPortRangeStart := int32(inboundNatPool["frontend_port_range_start"].(int))
+		frontendPortRangeEnd := int32(inboundNatPool["frontend_port_range_end"].(int))
+
+		networkSecurityGroupRules, err := ExpandBatchPoolNetworkSecurityGroupRule(inboundNatPool["network_security_group_rules"].([]interface{}))
+		if err != nil {
+			return nil, err
+		}
+
+		inboundNatPools[i] = batch.InboundNatPool{
+			Name:                      &name,
+			Protocol:                  protocol,
+			BackendPort:               &backendPort,
+			FrontendPortRangeStart:    &frontendPortRangeStart,
+			FrontendPortRangeEnd:      &frontendPortRangeEnd,
+			NetworkSecurityGroupRules: &networkSecurityGroupRules,
+		}
+	}
+
+	endpointConfiguration.InboundNatPools = &inboundNatPools
+
+	return endpointConfiguration, nil
+}
+
+//ExpandBatchPoolNetworkSecurityGroupRule expands Batch pool network security group rule
+func ExpandBatchPoolNetworkSecurityGroupRule(list []interface{}) ([]batch.NetworkSecurityGroupRule, error) {
+	if len(list) == 0 {
+		return nil, nil
+	}
+
+	networkSecurityGroupRule := make([]batch.NetworkSecurityGroupRule, len(list))
+
+	for i, groupRule := range list {
+		groupRuleMap := groupRule.(map[string]interface{})
+
+		priority := int32(groupRuleMap["priority"].(int))
+		sourceAddressPrefix := groupRuleMap["source_address_prefix"].(string)
+		access := batch.NetworkSecurityGroupRuleAccess(groupRuleMap["access"].(string))
+
+		networkSecurityGroupRule[i] = batch.NetworkSecurityGroupRule{
+			Priority:            &priority,
+			SourceAddressPrefix: &sourceAddressPrefix,
+			Access:              access,
+		}
+	}
+
+	return networkSecurityGroupRule, nil
+}
+
+// FlattenBatchPoolNetworkConfiguration flattens the network configuration for a Batch pool
+func FlattenBatchPoolNetworkConfiguration(networkConfig *batch.NetworkConfiguration) []interface{} {
+	results := make([]interface{}, 0)
+
+	if networkConfig == nil {
+		log.Printf("[DEBUG] networkConfgiuration is nil")
+		return nil
+	}
+
+	result := make(map[string]interface{})
+
+	if networkConfig.SubnetID != nil {
+		result["subnet_id"] = *networkConfig.SubnetID
+	}
+
+	if networkConfig.EndpointConfiguration != nil {
```

Comment:
if we shorten this here
```suggestion
	if cfg := networkConfig.EndpointConfiguration; cfg != nil {
```